state/state.go

```diff
@@ -300,6 +326,21 @@ func flagsForRelease(basePath string, release *ReleaseSpec) ([]string, error) {
 		}
 		flags = append(flags, "--values", valfileRendered)
 	}
+	for _, value := range release.Secrets {
+		valfile := filepath.Join(basePath, value)
+		path, err := renderTemplateString(valfile)
+		if err != nil {
+			return nil, err
+		}
+
+		valfileRendered, err := helm.SecretDecrypt(path)
+		if err != nil {
+			return nil, err
+		}
+
+		release.generatedSecrets = append(release.generatedSecrets, valfileRendered)
```

Comment:
I prefer calling it `generatedValues` rather than `generatedSecrets`, because what you append to it here is actually a `valfile` generated from a secrets file.